tap_github/streams.py

```diff
@@ -904,3 +904,46 @@ def http_headers(self) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class StargazersStream(GitHubStream):
+    """Defines 'Stargazers' stream."""
+
+    name = "stargazers"
+    path = "/repos/{org}/{repo}/stargazers"
+    primary_keys = ["repo", "org"]
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    replication_key = "starred_at"
```

Comment:
I'm not sure if this will capture removals, and if not, what do you think about disabling "incremental" on this stream by removing `replication_key`?

I'm not sure myself how to best handle this. In theory this is a relatively low-volume stream so full replication may be okay. But I can't say with confidence that the lack of deletion tracking is enough reason to fully remove incremental as an option.

Thoughts?

Reply:
I think it can actually be a quite heavy stream as soon as you look at big projects like https://github.com/facebook/react-native with 100k stars...

Unstarring events are probably not that common too. Maybe we can add a warning and people can drop their table from time to time. Or do you have other reset/purge mechanisms?